### PR TITLE
Update states.php -> Add Morocco regions

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add regions for Morocco according to CLDR 

https://github.com/unicode-org/cldr/blob/f800890ea86482c2eb5f73224897f4a8bc0b653a/common/subdivisions/en.xml#L3207

See https://airtable.com/appra4zMRgiGscWB7/tblqKcO8HXWoSFKUR/viwVKXWZNUe7840Lb/recBOhxl2CFVmOIGl?blocks=hide

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use Shipping Settings to test, `/wp-admin/admin.php?page=wc-settings&tab=shipping&zone_id=new`
2. Use the Region Picker to search for Morocco and ensure the results show the country and its regions. Note, the Region picker is new, so if you are testing this locally instead of on JN, be sure to run `pnpm build` from `plugins/woocommerce-admin`.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add Moroccan regions to states.php

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/i18n/states.php [hunk 1]
```diff
@@ -1145,6 +1145,102 @@
 		'SN' => __( 'Sinoe', 'woocommerce' ),
 	),
 	'LU' => array(),
+	'MA' => array( // Moroccan regions.
+			'ma01'  => __( 'Tangier-Tétouan', 'woocommerce' ),
```
Comment (coreymckrill) [inline plugins/woocommerce/i18n/states.php]: It looks like all the lines in this array have [one too many tab indents](https://github.com/woocommerce/woocommerce/actions/runs/7026425717/job/19152598723?pr=41769)...

**Review by coreymckrill — COMMENTED** [review]
There are some Playwright API tests failing here. At least one of them looks like it might be related (might need to add the Moroccan regions to the expected return value), but there's another one that doesn't seem related. So it might be worth rebasing this in addition to checking to see if something needs to be updated.

Sidenote: Probably not relevant for this particular PR, but I noticed that in the Region Picker the Moroccan states aren't listed in alphabetical order, which might make it harder to find a particular one when looking.

**Comment by psealock** [comment]
@peterfabian I've added Moroccan regions similar to your [PR here](https://github.com/woocommerce/woocommerce/pull/35199). ~Would you have any clue as to why CRUD tests would be failing?~ EDIT: Nevermind, I think were good here.

```
  7 failed
    [Chrome] › payment-gateways/payment-gateways-crud.test.js:11:2 › Payment Gateways API tests › can view all payment gateways 
    [Chrome] › products/product-list.test.js:2802:3 › Products API tests: List All Products › List all products › status 
    [Chrome] › products/products-crud.test.js:932:3 › Products API tests: CRUD › Product shipping classes tests: CRUD › can batch update product shipping classes 
    [Chrome] › products/products-crud.test.js:1414:2 › Products API tests: CRUD › can view a single product 
    [Chrome] › products/products-crud.test.js:1423:2 › Products API tests: CRUD › can update a single product 
    [Chrome] › products/products-crud.test.js:1443:2 › Products API tests: CRUD › can delete a product 
    [Chrome] › settings/settings-crud.test.js:276:3 › Settings API tests: CRUD › List all settings options › can retrieve all general settings 
  22 skipped
  232 passed (1.5m)
```

**Comment by psealock** [comment]
@coreymckrill Do you have a minute to give this another look? 🙇🏽‍♂️

**Review by coreymckrill — APPROVED** [review]
All looks good now! States are now listed in a better order too, thanks!

**Pull request merged**